api/src/routes/wallet.ts

```diff
@@ -0,0 +1,41 @@
+import z from 'zod'
+import { zValidator } from '@hono/zod-validator'
+import {
+  checkHrefFormat,
+  getWalletAddress,
+  normalizeWalletAddress,
+  toWalletAddressUrl,
+  WalletAddressFormatError
+} from '@shared/utils'
+import { app } from '../app.js'
+import { createHTTPException } from '../utils/utils.js'
+
+const walletAddressSchema = z.object({
+  walletAddress: z.url('Wallet address must be a valid URL')
+})
+
+app.get(
+  '/wallet',
+  zValidator('query', walletAddressSchema),
+  async ({ req, json }) => {
+    const { walletAddress } = req.valid('query')
+
+    try {
+      const walletAddressUrl = checkHrefFormat(
+        toWalletAddressUrl(walletAddress)
```

Comment:
I don't think we need `toWalletAddressUrl` (or even `checkHrefFormat`) here as input is a URL. This API is called internally only with particular inputs that we control, and it failing with bad input with fail at `getWalletAddress` itself.